cmd/blox_render.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright © 2021 NAME HERE <EMAIL ADDRESS>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	tpl "text/template"
+
+	"github.com/cueblox/blox"
+	"github.com/cueblox/blox/internal/cuedb"
+	"github.com/pterm/pterm"
+	"github.com/spf13/cobra"
+)
+
+var template string
+var with string
+var each bool
+
+// renderCmd represents the execute command
+var renderCmd = &cobra.Command{
+	Use:   "render",
+	Short: "Render templates with compiled data",
+	Long: `Render templates with compiled data. 
+Use the 'with' parameter to restrict the data set to a single content type.
+Use the 'each' parameter to execute the template once for each item.`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		// begin cut and paste from blox_build
+		userConfig, err := ioutil.ReadFile("blox.cue")
+
+		pterm.Debug.Printf("loading user config")
+
+		cobra.CheckErr(err)
+
+		engine, err := cuedb.NewEngine()
+
+		pterm.Debug.Printf("new engine")
+		cobra.CheckErr(err)
+
+		cfg, err := blox.NewConfig(BaseConfig)
+
+		pterm.Debug.Printf("newConfig")
+		cobra.CheckErr(err)
+
+		err = cfg.LoadConfigString(string(userConfig))
+		cobra.CheckErr(err)
+
+		// Load Schemas!
+		schemataDir, err := cfg.GetString("schemata_dir")
+		cobra.CheckErr(err)
+
+		remotes, err := cfg.GetList("remotes")
+		if err == nil {
+			parseRemotes(remotes)
+		}
+
+		pterm.Debug.Printf("\t\tUsing schemata from: %s\n", schemataDir)
+
+		err = filepath.WalkDir(schemataDir, func(path string, d fs.DirEntry, err error) error {
```

Comment:
Is it possible to actually call the build command, instead of duplicating?

Reply:
I tried to find a way, but couldn't see anything in the cobra docs.

Reply:
reference #40 - refactor commands to use a central logic structure.